File: Programming_Basics_with_Python/conditional_statements_advanced_exercise/04._fishing_boat.py
budget = int(input())
season = input()
fishers = int(input())
boat_price = 0
if season == "Spring":
    boat_price = 3000
    if fishers <= 6:
        boat_price -= 0.1 * boat_price
    elif 6 < fishers <= 11:
        boat_price -= 0.15 * boat_price
    elif fishers > 12:
        boat_price -= 0.25 * boat_price
elif season == "Summer" or season == "Autumn":
    boat_price = 4200
    if fishers <= 6:
        boat_price -= 0.1 * boat_price
    elif 6 < fishers <= 11:
        boat_price -= 0.15 * boat_price
    elif fishers > 12:
        boat_price -= 0.25 * boat_price
elif season == "Winter":
    boat_price = 2600
    if fishers <= 6:
        boat_price -= 0.1 * boat_price
    elif 6 < fishers <= 11:
        boat_price -= 0.15 * boat_price
    elif fishers > 12:
        boat_price -= 0.25 * boat_price
if fishers % 2 == 0 and season != "Autumn":
    discount = 0.05 * boat_price
    boat_price -= discount
if budget >= boat_price:
    diff = budget - boat_price
    print(f"Yes! You have {diff:.2f} leva left.")
else:
    diff = boat_price - budget
    print(f"Not enough money! You need {diff:.2f} leva.")

File: Programing_Basics_with_JS/conditionalStatementsAdvancedExercise/04.FishingBoat.js
function fishingBoat(input) {
    let groupBudget = Number(input[0]);
    let season = input[1];
    let fishermanCount = Number(input[2]);

    let boatRent = 0;

    switch(season) {
        case 'Spring': boatRent = 3000; break;
        case 'Summer': 
        case 'Autumn': boatRent = 4200; break;
        case 'Winter': boatRent = 2600; break;
    }

    if (fishermanCount <= 6) {
        boatRent = boatRent * 0.9;
    } else if (fishermanCount <= 11) {
        boatRent = boatRent * 0.85;
    } else {
        boatRent = boatRent * 0.75;
    }

    if (fishermanCount % 2 == 0 && season !== 'Autumn') {
        boatRent = boatRent * 0.95;
    }

    let moneyDiff = Math.abs(groupBudget - boatRent);

    if (groupBudget >= boatRent) {
        console.log(`Yes! You have ${moneyDiff.toFixed(2)} leva left.`);
    } else {
        console.log(`Not enough money! You need ${moneyDiff.toFixed(2)} leva.`);
    }
}

fishingBoat(["3000",
"Summer",
"11"])

fishingBoat(["3600",
"Autumn",
"6"])

fishingBoat(["2000",
"Winter",
"13"])
